src/components/pop-up/index.tsx

```diff
@@ -0,0 +1,57 @@
+import styled from '@emotion/styled'
+import React from 'react'
+
+type Props = {
+  children: React.ReactNode
+  placeInView: [
+    isDisplayed: boolean,
+    setDisplay: React.Dispatch<React.SetStateAction<boolean>>,
+  ]
+}
+
+function PopUp({ children, placeInView }: Props) {
+  const [isDisplayed, setDisplay] = placeInView
+  const clickListenerRef = React.useRef<HTMLDivElement | null>(null)
+
+  React.useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (
+        clickListenerRef.current &&
+        clickListenerRef.current !== event.target
+      ) {
+        setDisplay(false)
+      }
+    }
+    document.addEventListener('click', handleClickOutside, true)
+    return () => {
+      document.removeEventListener('click', handleClickOutside, true)
+    }
+  })
```

Comment:
this run on every render, is this intentional? Also it's best to try and avoid `useEffect`  when possible since it's prone mistakes

Reply:
but we will need that if someone clicks outside the popup or is there a better way to achieve the same outcome.

Reply:
[borrowing from Material-UI.ClickAwayListener](https://github.com/mui-org/material-ui/blob/master/packages/material-ui/src/ClickAwayListener/ClickAwayListener.js) first we learn that correct popper behavior is complicated, but we can see that registering the listener once is enough per click-handler iteration, no need to re-register per-each-render. The short fix placing `handleClickOutside` -- outside the effect, in a `useCallback` and placing that as a depenency
```ts
const handleClickOutside = React.useCallback<MouseEventHandler<SomeHTMLElement>>(e => {}, [])

React.useEffect( () => {
  document.addEventListener('click', handleClickOutside, {capture: true))
  return () => {
    document.removeEventListener('click', handleClickOutside, {capture: true})
  }
}, [handleClickOutside]
```
(note the use of the option object instead of passing a boolean. No one can remember the entire browser API, so this helps hint at what is going on)

Reply:
I don't think `useCallback` is necessary here either. The original code is also a lot easier to read IMO.

Reply:
this is a potential runaway useEffect, regardless of if and where you place a `useCallback`. This will register and re-register many many times potentially.